sql/expression/div.go

```diff
@@ -389,6 +389,14 @@ func getFloatOrMaxDecimalType(e sql.Expression, treatIntsAsFloats bool) sql.Type
 					maxFrac = s
 				}
 			}
+		case *Convert:
+			p, s := GetPrecisionAndScale(c.cachedDecimalType)
```

Comment:
Are we sure `c.cachedDecimalType` will be populated? Might be good to add a new test for some non-numeric types, like:
`SELECT CASE WHEN COUNT( * ) THEN 10 * CAST(1234 AS DATE) + CAST(82 AS CHAR) END;`

Reply:
Added a nil check and tests for this case